src/OgFieldBase.php

```diff
@@ -29,9 +69,30 @@
    * @param String $entity_type
    *   The entity type.
    *
-   * @return OgFieldBase
+   * @return \Drupal\og\OgFieldBase
+   * @throws \Exception
+   *   Throw error if the field storage config definition explicitly defines to
+   *   which entities the field can be attached to.
    */
   public function setEntityType($entity_type) {
+    $field_storage = $this->getFieldStorageConfigBaseDefinition();
+
+    if (!empty($field_storage['entity']) && !in_array($entity_type, $field_storage['entity'])) {
+
+      $params = [
+        '@plugin' => $this->getPluginId(),
+        '@entity' => implode(', ', $field_storage['entity']),
+      ];
+
+      if ($field_name = $this->getFieldName()) {
+        $params['@field_name'] = $field_name;
+        throw new \Exception(new FormattableMarkup('The Organic Groups field with plugin ID @plugin with the name @field_name cannot be attached to the entity type. It can only be attached to the following entities: @entity.', $params));
+      }
+
+      throw new \Exception(new FormattableMarkup('The Organic Groups field with plugin ID @plugin cannot be attached to the entity type. It can only be attached to the following entities: @entity.', $params));
```

Comment:
This is rather interesting. It is possible that a field doesn't have its field name set because we manually have to call the setters for field name, bundle and entity type when the field definition is initially retrieved in `Og::getFieldBaseDefinition()`. This is not very convenient in this particular scenario since we have to check whether the field name is set in order to generate a correct exception message, but it's probably never going to bother anyone, since everybody will simply call `Og::createField()` which takes care of it.

What probably WILL bother people is that the `::getFieldName()` method is public but is not part of `OgFieldsInterface` so you cannot even trust that this method is available. We should add the getters and setters for the field name, entity type and bundle to the interface.